deploy/helm-chart/templates/secret-connection.yaml

```diff
@@ -17,11 +17,29 @@ stringData:
   {{- if ne (.Values.connection.uri | toString) "" }}
   PROMSCALE_DB_URI: {{ tpl .Values.connection.uri . | toString | quote }}
   {{- else }}
-  PROMSCALE_DB_PORT: {{ .Values.connection.port | toString | quote }}
-  PROMSCALE_DB_USER: {{ .Values.connection.user | toString | quote }}
+  {{- /* password and user can be set conditionally to allow potential password injection from other sources */}}
+  {{- if ne (.Values.connection.password | toString) "" }}
   PROMSCALE_DB_PASSWORD: {{ .Values.connection.password | toString | quote }}
+  {{- end }}
+  {{- if ne (.Values.connection.user | toString) "" }}
```

Comment:
The default user is values.yml is not empty string. I think we need to change that for this to work.